Update docs to require Gradle 4.9

### Expected behavior
Expect this to work with Gradle 4.3

### Actual behavior
```
Creating deblibs tasks...


FAILURE: Build failed with an exception.

* What went wrong:
org.gradle.api.tasks.TaskContainer.register(Ljava/lang/String;Ljava/lang/Class;Lorg/gradle/api/Action;)Lorg/gradle/api/tasks/TaskProvider;
```

### Steps to reproduce the behavior
 - Install gradle 4.3
 - run `./gradlew --stacktrace --info createSlackMessage`

From the gradle docs, it looks like this endpoint was added in 4.9:
https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/TaskContainer.html#register-java.lang.String-org.gradle.api.Action-

--- a/.gitignore
+++ b/.gitignore
@@ -5,3 +5,4 @@ build/
 !gradle-wrapper.jar
 *.iml
 local.properties
+out/

--- a/README.md
+++ b/README.md
@@ -3,7 +3,7 @@
 DebLibs
 -------
 
-A Gradle plugin that creates Github issue and Slack message for outdated project dependencies so they 
+A Gradle plugin that creates Github issue and Slack message for outdated project dependencies so they
 can easily be tracked and manually upgraded.
 
 This plugin builds on the [Gradle Versions Plugin](https://github.com/ben-manes/gradle-versions-plugin).
@@ -45,11 +45,11 @@ buildscript {
   dependencies {
     classpath "com.hellofresh.gradle:deblibs:$version"
   }
-  
+
 }
 ```
 
-The plugin show work with Gradle version 4.3 and above.
+The plugin show work with Gradle version 4.9 and above.
 
 Configuration
 -------------
@@ -80,7 +80,7 @@ The second task is for publishing info about outdated dependencies to a Slack ch
 
 To create a Github issue, issue the command:
 
-`./gradlew createGithubIssue` 
+`./gradlew createGithubIssue`
 
 To post to Slack, issue the command:
 

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -102,10 +102,10 @@ setupPublishingEnvironment()
 
 dependencies {
     testImplementation("io.kotlintest:kotlintest-runner-junit5:3.1.9")
-    implementation("com.github.ben-manes:gradle-versions-plugin:0.20.0")
-    implementation("com.squareup.okhttp3:okhttp:3.8.1")
-    implementation("com.squareup.moshi:moshi:1.7.0")
-    implementation("com.squareup.okio:okio:2.1.0")
+    implementation("com.github.ben-manes:gradle-versions-plugin:0.27.0")
+    implementation("com.squareup.okhttp3:okhttp:4.3.1")
+    implementation("com.squareup.moshi:moshi:1.9.2")
+    implementation("com.squareup.okio:okio:2.4.3")
     implementation(kotlin("stdlib"))
 }
 

--- a/src/main/kotlin/com/hellofresh/deblibs/BaseClient.kt
+++ b/src/main/kotlin/com/hellofresh/deblibs/BaseClient.kt
@@ -19,7 +19,7 @@ package com.hellofresh.deblibs
 import okhttp3.MediaType.Companion.toMediaTypeOrNull
 import okhttp3.OkHttpClient
 import okhttp3.Request
-import okhttp3.RequestBody
+import okhttp3.RequestBody.Companion.toRequestBody
 import okhttp3.Response
 
 interface BaseClient : Runnable {
@@ -43,7 +43,7 @@ interface BaseClient : Runnable {
         headerName: String,
         headerValue: String
     ): Pair<Response?, Int> {
-        val requestBody = RequestBody.create(JSON, json)
+        val requestBody = json.toRequestBody(JSON)
         val request = requestBuilder.addHeader(headerName, headerValue)
             .url(url)
             .post(requestBody)
@@ -53,7 +53,7 @@ interface BaseClient : Runnable {
 
         try {
             response = client.newCall(request).execute()
-            status = response.code()
+            status = response.code
         } finally {
             response?.close()
         }

--- a/src/main/kotlin/com/hellofresh/deblibs/BaseClient.kt
+++ b/src/main/kotlin/com/hellofresh/deblibs/BaseClient.kt
@@ -16,7 +16,7 @@
 
 package com.hellofresh.deblibs
 
-import okhttp3.MediaType
+import okhttp3.MediaType.Companion.toMediaTypeOrNull
 import okhttp3.OkHttpClient
 import okhttp3.Request
 import okhttp3.RequestBody
@@ -27,7 +27,7 @@ interface BaseClient : Runnable {
     companion object {
 
         val client = OkHttpClient()
-        val JSON = MediaType.parse("application/json; charset=utf-8")
+        val JSON = "application/json; charset=utf-8".toMediaTypeOrNull()
         const val AUTHORIZATION = "Authorization"
         val requestBuilder: Request.Builder
             get() {

--- a/src/main/kotlin/com/hellofresh/deblibs/github/GithubClient.kt
+++ b/src/main/kotlin/com/hellofresh/deblibs/github/GithubClient.kt
@@ -19,12 +19,7 @@ package com.hellofresh.deblibs.github
 import com.hellofresh.deblibs.Adapters
 import com.hellofresh.deblibs.BaseClient
 import com.hellofresh.deblibs.BaseClient.Companion.AUTHORIZATION
-import com.hellofresh.deblibs.BaseClient.Companion.JSON
-import com.hellofresh.deblibs.BaseClient.Companion.client
-import com.hellofresh.deblibs.BaseClient.Companion.requestBuilder
 import com.squareup.moshi.JsonAdapter
-import okhttp3.Request
-import okhttp3.RequestBody.Companion.toRequestBody
 import okhttp3.Response
 import java.net.HttpURLConnection
 

--- a/src/main/kotlin/com/hellofresh/deblibs/github/GithubClient.kt
+++ b/src/main/kotlin/com/hellofresh/deblibs/github/GithubClient.kt
@@ -19,7 +19,12 @@ package com.hellofresh.deblibs.github
 import com.hellofresh.deblibs.Adapters
 import com.hellofresh.deblibs.BaseClient
 import com.hellofresh.deblibs.BaseClient.Companion.AUTHORIZATION
+import com.hellofresh.deblibs.BaseClient.Companion.JSON
+import com.hellofresh.deblibs.BaseClient.Companion.client
+import com.hellofresh.deblibs.BaseClient.Companion.requestBuilder
 import com.squareup.moshi.JsonAdapter
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.toRequestBody
 import okhttp3.Response
 import java.net.HttpURLConnection
 
@@ -41,7 +46,7 @@ class GithubClient(
             if (status == HttpURLConnection.HTTP_NOT_FOUND) {
                 error("404 Repository at '$repo' was not found")
             }
-            error("Could not create github issue: $status ${response?.message()}\n$githubIssue")
+            error("Could not create github issue: $status ${response?.message}\n$githubIssue")
         }
     }
 

--- a/src/main/kotlin/com/hellofresh/deblibs/gitlab/GitlabClient.kt
+++ b/src/main/kotlin/com/hellofresh/deblibs/gitlab/GitlabClient.kt
@@ -40,7 +40,7 @@ class GitlabClient(
             if (status == HttpURLConnection.HTTP_NOT_FOUND) {
                 error("404 Repository at '$id' was not found")
             }
-            error("Could not create github issue: $status ${response?.message()}\n$gitlabIssue")
+            error("Could not create github issue: $status ${response?.message}\n$gitlabIssue")
         }
     }
 

--- a/src/main/kotlin/com/hellofresh/deblibs/slack/SlackClient.kt
+++ b/src/main/kotlin/com/hellofresh/deblibs/slack/SlackClient.kt
@@ -20,6 +20,8 @@ import com.hellofresh.deblibs.Adapters.adapter
 import com.hellofresh.deblibs.BaseClient
 import com.hellofresh.deblibs.BaseClient.Companion.AUTHORIZATION
 import com.squareup.moshi.JsonAdapter
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.toRequestBody
 import okhttp3.Response
 
 class SlackClient(

--- a/src/main/kotlin/com/hellofresh/deblibs/slack/SlackClient.kt
+++ b/src/main/kotlin/com/hellofresh/deblibs/slack/SlackClient.kt
@@ -20,8 +20,6 @@ import com.hellofresh.deblibs.Adapters.adapter
 import com.hellofresh.deblibs.BaseClient
 import com.hellofresh.deblibs.BaseClient.Companion.AUTHORIZATION
 import com.squareup.moshi.JsonAdapter
-import okhttp3.Request
-import okhttp3.RequestBody.Companion.toRequestBody
 import okhttp3.Response
 
 class SlackClient(
@@ -39,7 +37,7 @@ class SlackClient(
         val (response, status) = makePostRequest()
         val isSuccessful = response?.isSuccessful ?: return
         if (!isSuccessful) {
-            error("Could not create slack message: $status ${response.message()}\n")
+            error("Could not create slack message: $status ${response.message}\n")
         }
     }
 

--- a/src/main/kotlin/com/hellofresh/deblibs/slack/SlackMessage.kt
+++ b/src/main/kotlin/com/hellofresh/deblibs/slack/SlackMessage.kt
@@ -36,7 +36,7 @@ data class SlackMessage(
         enum class Color(val hexColor: String) {
             RED("#c4291c"),
             GREEN("#36a64f"),
-            YELLOW("#d6a048")
+            YELLOW("#d6a048"),
         }
     }
 

